### ISSUE TITLE
Improve engine

### ISSUE THREAD

**Issue by cajoy** [hjr265/too#1]
Let's say I want to add 50 likes in a sec.

Suggestions and Similarities are going to be updated 50 times during query.

Can I somehow manually run update after I added 50 likes.

Because current implementation is going to burn CPU

Thanks

**Comment by hjr265** [comment]
@cajoy You are absolutely right! To be honest, the current implementation is far from ideal, or at least what I want it to be.

I was planning to resume working on this very very soon, and I will probably start with this suggestion of yours. Thank you!

**Comment by cajoy** [comment]
Sounds good for me. I can try to help if you don't mind..

**Comment by hjr265** [comment]
Contributions are always welcome :)

Just, if possible, keep me in the loop before preparing a pull request, so that we don't end up duplicating efforts.

**Comment by cajoy** [comment]
ok
